src/funexpr.jl

```diff
@@ -30,102 +30,47 @@ function sanitize(ref::GlobalRef)
-                                 "with types $types, expected exactly 1 method")
-        lambda = ms[1].lambda_template
-        slot_ex_arr = Base.uncompressed_ast(lambda)
-        slot_ex = sanitize(Expr(:block, slot_ex_arr...))
-        slotnames = lambda.slotnames
-        ex = replace_slots(slot_ex, slotnames)
-        # 1st arg is a function name, next `lambda.nargs-1` are actual arg names
-        args = map(Symbol, slotnames[2:lambda.nargs])
-        return args, sanitize(ex)
-    end
-
-else
-
-    function replace_slots(ex::Expr, slotnames::Vector)
-        new_args = Array{Any}(length(ex.args))
-        for (i, arg) in enumerate(ex.args)
-            if isa(arg, Slot)
-                new_args[i] = slotnames[arg.id]
-            elseif isa(arg, Expr)
-                new_args[i] = replace_slots(arg, slotnames)
-            else
-                new_args[i] = arg
-            end
+function replace_slots(ex::Expr, slotnames::Vector)
+    new_args = Array{Any}(length(ex.args))
+    for (i, arg) in enumerate(ex.args)
+        if isa(arg, Slot)
+            new_args[i] = slotnames[arg.id]
+        elseif isa(arg, Expr)
+            new_args[i] = replace_slots(arg, slotnames)
+        else
+            new_args[i] = arg
         end
-        new_ex = Expr(ex.head, new_args...)
-        return new_ex
     end
-
-    function arg_names(sig::Expr)
-        return [isa(arg,  Symbol) ? arg : arg.args[1] for arg in sig.args[2:end]]
+    new_ex = Expr(ex.head, new_args...)
+    return new_ex
+end
+function arg_names(sig::Expr)
+    return [isa(arg,  Symbol) ? arg : arg.args[1] for arg in sig.args[2:end]]
+end
+function to_expr(src::CodeInfo)
+    if isa(src.code, Array{Any,1})
+        slotnames = [Symbol(name) for name in Base.sourceinfo_slotnames(src)]
+        body = Expr(:body, src.code...)
+        result = replace_slots(body, slotnames)
+        return result
+    else
+        error("Can't convert CodeInfo to expression: CodeInfo is compressed: $src")
     end
-
-
-    function to_expr(src::CodeInfo)
-        if isa(src.code, Array{Any,1})
-            slotnames = [Symbol(name) for name in Base.sourceinfo_slotnames(src)]
-            body = Expr(:body, src.code...)
-            result = replace_slots(body, slotnames)
-            return result
+end
+function funexpr(f::Function, types::NTuple{N,DataType}) where N
+    method = Sugar.get_method(f, types)
+    file = string(method.file)
+    linestart = method.line
+    try
+        code, _ = Sugar.get_source_at(file, linestart)
+        return arg_names(code.args[1]), sanitize(code.args[2])
+    catch err
+        if isa(err, LoadError)
+            code = code_lowered(f, types)[1]
+            args = convert(Vector{Symbol}, code.slotnames[2:end])
+            return args,to_expr(code)
         else
-            error("Can't convert CodeInfo to expression: CodeInfo is compressed: $src")
+            rethrow(err)
```

Comment:
You need to make the comment with "changes requested", otherwise the bot ignores this. In any case, could you explain what the issue is? GitHub has a bit of trouble with the diff, but this looks correct to me. Am I missing an issue?